bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/UpdateSecretCommand.cs

```diff
@@ -30,17 +30,29 @@ public async Task<Secret> UpdateAsync(Secret updatedSecret, Guid userId)
 
         var orgAdmin = await _currentContext.OrganizationAdmin(secret.OrganizationId);
         var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+        var originalProject = secret.Projects?.FirstOrDefault();
 
-        var project = updatedSecret.Projects?.FirstOrDefault();
-
-        var hasAccess = accessClient switch
+        var hasAccessToOriginalProject = accessClient switch
```

Comment:
I think we could clean this up by extracting it to a function and writing

```c#
switch (accessClient) {
  case AccessClientType.NoAccessCheck:
    return true;
  case AccessClientType.User:
    var oldProject = secret.Projects?.FirstOrDefault();
    var newProject = updatedSecret.Projects?.FirstOrDefault();

    var accessToOld = originalProject != null && await _projectRepository.UserHasWriteAccessToProject(originalProject.Id, userId);
    var accessToNew = newProject != null && await _projectRepository.UserHasWriteAccessToProject(newlyAssignedProject.Id, userId);

    return accessToOld && accessToNew;
  default:
    return false;
}
```

Reply:
Sounds like a good idea, thanks Oscar!